### ISSUE TITLE
Suggest refreshing the browser when server connection is lost to a server-side Blazor app

### ISSUE THREAD

**Issue by danroth27** [aspnet/AspNetCore#10496]
Currently we provide a Retry button when the connection to a server-side Blazor app is lost and could not be automatically reestablished. 

![image](https://user-images.githubusercontent.com/1874516/58277439-9a3b2800-7d4e-11e9-847e-1ea5dd8a9ffc.png)

In many cases retrying is not going to work because the server has been restarted and the circuit was lost (e.g. during development).  Instead the user needs to refresh the browser. We should update the UI to suggest that the user do so. Maybe just add a sentence below that says: "If you are unable to reconnect to the server try refreshing the browser"?

**Comment by monstergarden** [comment]
Why not make the "retry?" button refresh the page? 
window.location.reload(false);

**Comment by mkArtakMSFT** [comment]
The work here is about updating the UI shown above, to include text suggesting to try to refresh the page, if `Retry` doesn't work.

**Comment by danroth27** [comment]
@monstergarden Retry should work after a momentary network connection loss, so I think having the Retry button is still valid.

**Comment by kjeske** [comment]
Could we have that text configurable, so developers can decide themselves what to put into the alert? It would be even nicer if we could customize the whole content of that element with id "components-reconnect-modal". The alert behavior could be customized in options passed the Blazor.start JS function.

Example:
```js
var blazor = window.Blazor;

blazor.start({
    getConnectionIssueElement: function () {
        var element = document.createElement("div");
        [...] // create content, attach events to retry
        return element;
    }
});
```

Any kind of customization of the alert would be nice.

**Comment by danroth27** [comment]
@kjeske You can already customize this UI today. See https://docs.microsoft.com/aspnet/core/blazor/hosting-models#reconnection-to-the-same-server for details.

**Comment by willdean** [comment]
We are using server-side Blazor to run a front-panel (touch-only) on a bit of equipment.   If we upgrade the server-side code (we can do this behind the scenes) we'd like the client to reconnect by whatever means possible - we don't find the 'retry' button ever does anything useful/visible in any situation we've seen (neither VS background compile, nor 'real system' upgrade), and without connecting an external keyboard we can't do any other kind of browser refresh.

We've done the following, which seems to work OK so far, but it would be nice if a 'just silently reconnect' option was available.

```
setInterval(function() {
                var modal = document.getElementById("components-reconnect-modal");
                if (modal.classList.contains("components-reconnect-failed")) {
                    window.location.reload(false);
                }
            },
            1000);
```

**Comment by danroth27** [comment]
@willdean I'd be interested to hear more details about what you're building. Shoot me an email (daroth@microsoft.com) if you have a moment!

**Comment by dedmen** [comment]
Can't the server tell the client that his circuit is not available anymore?
How about the client just starting with a fresh new circuit or just doing a site reload if the code notices that connecting to the old circuit is not working?

I made my blazor app into a progressive web app, and everytime I turn the phone screen off, and back on after a few minutes it will show the failed to reconnect. Because it disconnected while the screen was off, and the server cleaned up the circuit.

**Comment by SteveSandersonMS** [comment]
@dedmen Yes, that's what we're planning in https://github.com/aspnet/AspNetCore/issues/12442

In the meantime, closing this issue as implemented in #12420
